### ISSUE TITLE
Loss function 

### ISSUE THREAD

**Issue by ktran1** [pietz/simclr#1]
Hi, thanks for your implementation. I wonder if there is a minor error for the loss function.  
Maybe torch.mean(-torch.log(x)) instead of -torch.log(x.mean()) ?

**Comment by pietz** [comment]
Hey @ktran1 , thanks for your feedback. I'll check it more closely tonight but I believe the loss needs to be a scalar and not a sample-wise vector. I'll get back to you.

**Comment by pietz** [comment]
Yes, as expected applying your change throws an error because the loss needs to be a scalar.

The loss function formula in the paper (F. 1) calculates the loss for a single sample from the batch. So we need to take the mean of it: `(-torch.log(x)).mean()`. This is equivalent to `-torch.log(x.mean())` from my implementation.
